src/App/Fossa/VSI/DynLinked/Types.hs

```diff
@@ -0,0 +1,30 @@
+module App.Fossa.VSI.DynLinked.Types (
+  LinuxPackageManager (..),
+  LinuxPackageMetadata (..),
+  DynamicDependency (..),
+) where
+
+import Data.Text (Text)
+import Path (Abs, File, Path)
+
+data LinuxPackageManager
+  = LinuxPackageManagerDEB
+  | LinuxPackageManagerRPM
+  | LinuxPackageManagerAPK
+
+-- | These are all opaque blobs, we're not doing any processing on them other than rendering into a @Locator@.
+-- For that reason, stick to @Text@.
+data LinuxPackageMetadata = LinuxPackageMetadata
+  { linuxPackageFetcher :: Text
+  , linuxPackageID :: Text
+  , linuxPackageRevision :: Text
+  , linuxPackageDistro :: Text
+  , linuxPackageDistroRelease :: Text
+  , linuxPackageArch :: Text
```

Comment:
This isn't that important since these types aren't used yet, but are we really guaranteed to find this much information for DEB, RPM, and APK deps? That seems impressive.

Reply:
Yes! Well, mostly. The important bit is any package that can't provide us all the required information "falls through" to be a user-defined dependency. In general though we expect most packages to be able to provide us all this information.